### PR TITLE
consume support for scheduler plugin informerMode from upstream

### PR DESCRIPTION
**Pull request by rbaturov**
Previously, the NUMA-aware scheduler plugin had d/s-specific code to enable the separate informer, which was in turn needed to get the full pod list vs the filtered pod list the scheduler framework provides. Thus, we had d/s specific enablement in the operator. Since commit [93c518b](https://github.com/kubernetes-sigs/scheduler-plugins/pull/599) the upstream added support for the separate informer, consumed by the KNI scheduler since commit [780ea84](https://github.com/openshift-kni/scheduler-plugins/pull/161). Remove the d/s specific enablement and set the proper configuration value.

We expect no changes from UX perspective. This is an internal change only.

### CODE REVIEW

**Comment by Tal-or** [comment]
@rbaturov retest won't help here
You haver some linter issues:
```
pkg/objectupdate/sched/sched.go:73: unnecessary trailing newline (whitespace)
```

**Comment by ffromani** [comment]
> This PR introduces changes required to fix the numaresources operator to consume the upstreamed separate informer support. Now, the informerMode field is passed through the topology-aware scheduler config map as expected from the scheduler to consume. The previous implementation of passing the informerMode field through the environment variable is removed.

This text is good, but still heavily skewed about the "who". IOW, it is summarizing the changes in plain english. This is not the content we should aim for. We should aim for text which describes the "why", giving context when needed to understand "why", why now? why is important? why is better than before (for a given definition of "better")?

We will (or we should...) go back and read git log and/or PR description to learn the motive and remember the context, not to get a quick grasp of the changes.

So what about something like
"Previously, the NUMA-aware scheduler plugin had d/s-specific code to enable the separate informer, which was in turn needed to get the full pod list vs the filtered pod list the scheduler framework provides. Thus, we had d/s specific enablement in the operator. Since commit XXX (TODO: figure it out) the upstream added support for the separate informer, consumed by the KNI scheduler since commit YYY (TODO: figure out). Remove the d/s specific enablement and set the proper configuration value.

We expect no changes from UX perspective. This is an internal change only."

File: controllers/numaresourcesscheduler_controller.go [hunk 1]
```diff
@@ -259,7 +259,7 @@ func configParamsFromSchedSpec(schedSpec nropv1.NUMAResourcesSchedulerSpec, cach
 	}
 	params.Cache.ResyncMethod = &resyncMethod
 	params.Cache.ForeignPodsDetectMode = &foreignPodsDetect
-
+
```
Comment (ffromani) [inline controllers/numaresourcesscheduler_controller.go]: unnecessary change, please remove

File: controllers/numaresourcesscheduler_controller.go [hunk 2]
```diff
@@ -259,7 +259,7 @@ func configParamsFromSchedSpec(schedSpec nropv1.NUMAResourcesSchedulerSpec, cach
 	}
```
Comment (ffromani) [inline controllers/numaresourcesscheduler_controller.go:260]: the commit message is empty here, but this is one of the rare cases on which is OK.
For extra point you can hint (1-2 lines) which feature you want to consume which prompted the bump, but the reason why is already pretty obvious from the PR context, so minimalistic commit message is fine.

File: controllers/numaresourcesscheduler_controller.go [hunk 3]
```diff
@@ -252,14 +252,20 @@ func configParamsFromSchedSpec(schedSpec nropv1.NUMAResourcesSchedulerSpec, cach
 
 	var foreignPodsDetect string
```
Comment (ffromani) [inline controllers/numaresourcesscheduler_controller.go:253]: please expand the commit message to explain succintly the "why" of this change. Feel free to take parts of the PR description and paste them here (or, equivalently, the PR description should build and expand on the content of the most important commit message(s) of the series, which in this case is this one).

File: controllers/numaresourcesscheduler_controller_test.go [hunk 4]
```diff
@@ -342,10 +342,6 @@ var _ = ginkgo.Describe("Test NUMAResourcesScheduler Reconcile", func() {
 					Name:  schedupdate.PFPStatusDumpEnvVar,
```
Comment (ffromani) [inline controllers/numaresourcesscheduler_controller_test.go:342]: this commit should be squashed in the previous

File: pkg/objectupdate/sched/sched_test.go [hunk 5]
```diff
@@ -73,10 +73,6 @@ var dpAllOptions = &appsv1.Deployment{
 								Name:  "PFP_STATUS_DUMP",
 								Value: "/run/pfpstatus",
 							},
-							{
```
Comment (ffromani) [inline pkg/objectupdate/sched/sched_test.go:76]: this change is good. I think we do want to preserve the intent of the tests, so check that we can effectively change the setting, using the new way (ConfigParams), don't we?

**Comment by ffromani** [comment]
Thanks for the PR! I left a few comments to improve over this good start. I'll wait for further review from both @Tal-or and @shajmakh !

**Comment by ffromani** [comment]
no point in further retests until the quick and basic lanes (e.g. CI Linter) are happy

**Comment by ffromani** [comment]
/retitle consume support for scheduler plugin informerMode from upstream

**Comment by ffromani** [comment]
bad rebase it seems?

**Comment by ffromani** [comment]
needs to be backported down to 4.14

**Comment by Tal-or** [comment]
Please add the CNF ticket in the PR header in order to sync it with Jira

**Comment by ffromani** [comment]
> Please add the CNF ticket in the PR header in order to sync it with Jira

I removed it. I'm unconvinced we should have this synchronization.

**Comment by Tal-or** [comment]
> > Please add the CNF ticket in the PR header in order to sync it with Jira
> 
> I removed it. I'm unconvinced we should have this synchronization.

This is mostly for stakeholders which are monitoring mainly with Jira so they'll have an easy way to associate the ticket with the PR

**Comment by ffromani** [comment]
> > > Please add the CNF ticket in the PR header in order to sync it with Jira
> > 
> > 
> > I removed it. I'm unconvinced we should have this synchronization.
> 
> This is mostly for stakeholders which are monitoring mainly with Jira so they'll have an easy way to associate the ticket with the PR

The question is if this project accepts these kinds of tags pointing to trackers external to the project.

File: controllers/numaresourcesscheduler_controller_test.go [hunk 6]
```diff
@@ -417,7 +410,55 @@ var _ = ginkgo.Describe("Test NUMAResourcesScheduler Reconcile", func() {
...
+
+			key := client.ObjectKeyFromObject(nrs)
+			_, err := reconciler.Reconcile(context.TODO(), reconcile.Request{NamespacedName: key})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			expectCacheParams(reconciler.Client, depmanifests.CacheResyncAutodetect, depmanifests.CacheResyncOnlyExclusiveResources, depmanifests.CacheInformerShared)
+
+		})
+
+		ginkgo.It("should allow to change the informerMode to Dedicated", func() {
```
Comment (ffromani) [inline controllers/numaresourcesscheduler_controller_test.go:444]: this test makes sense only if we
1. start with informerMode dedicated
2. we can correctly change to shared
3. we can then correctly change it back again to dedicated

if you followed another test which doesn't do that, the other test is wrong and needs to be fixed in another PR

File: pkg/objectupdate/sched/sched_test.go [hunk 7]
```diff
@@ -328,9 +324,6 @@ func TestDeploymentEnvVarSettings(t *testing.T) {
 	cacheResyncDebugEnabled := nropv1.CacheResyncDebugDumpJSONFile
 	cacheResyncDebugDisabled := nropv1.CacheResyncDebugDisabled
 
-	schedInformerShared := nropv1.SchedulerInformerShared
-	schedInformerDedicated := nropv1.SchedulerInformerDedicated
-
```
Comment (ffromani) [inline pkg/objectupdate/sched/sched_test.go:333]: why was this removed?

**Review by ffromani — COMMENTED** [review]
/lgtm

minor comments can be addressed in a later PR

**Comment by ffromani** [comment]
/approve

we need to do this work to catch up with scheduler-plugins code upstream

**Comment by ffromani** [comment]
let's start the backports once this code merges

**Comment by ffromani** [comment]
sweet
